### PR TITLE
Task #215730 : Integration of student basic details and fixed alignment of UI

### PR DESCRIPTION
**Pull request by rahulg1254**
basicD3

## Summary by CodeRabbit

- **New Features**
	- Enhanced the student details page with updated information and styling.
- **Bug Fixes**
	- Fixed the initial state of user data to prevent errors on load.
- **Refactor**
	- Optimized data fetching for student details using `useEffect`.
	- Improved code readability and maintainability in the student details component.
- **Style**
	- Updated the visual presentation of the student details page for a better user experience.

### CODE REVIEW

File: src/pages/StudentBasicDetails.tsx [hunk 1]
```diff
@@ -184,46 +169,18 @@ const StudentDetails = () => {
...
-              <Typography
-                sx={{
-                  color: theme.palette.text.primary,
-                  fontSize: '16px',
-                  fontWeight: 500
-                }}
-              >
-                {item.grade}
-              </Typography>
+              <Typography sx={{ fontSize: '16px', fontWeight: 500 }}>{userData?.dob}</Typography>
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:183]: @rahulg1254 I think 16px is default fontsize no need to specify explicitely

File: src/pages/StudentBasicDetails.tsx [hunk 2]
```diff
@@ -63,48 +77,32 @@ const StudentDetails = () => {
                   fontWeight: 500
                 }}
                 variant="h6"
-                gutterBottom
-              >
+                gutterBottom>
                 Attendance Report
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:81]: @rahulg1254 add translation

File: src/pages/StudentBasicDetails.tsx [hunk 3]
```diff
@@ -1,60 +1,74 @@
...
       <Header />
-      <Box mt={3} display={'flex'} gap={2} alignItems={'flex-start'}>
-        <Link to={'/'}>
+      <Box mt={3} display="flex" gap={2} alignItems="flex-start">
+        <Link to="/">
           <ArrowBackIcon sx={{ color: theme.palette.warning['A200'], fontSize: 'large' }} />
         </Link>
         <Stack>
-          <Typography fontSize={'22px'} variant="h3" fontFamily={'Poppins'}>
+          <Typography fontSize="22px" variant="h3" fontFamily="Poppins">
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:58]: @rahulg1254 no need to add fontFamily this level, we have added in the global level

File: src/pages/StudentBasicDetails.tsx [hunk 4]
```diff
@@ -1,60 +1,74 @@
...
+      EmploymentStatus: 'Employment Status'
     }
   ]);
 
-  const renderStatsCard = (label1: string, value1: string) => (
+  useEffect(() => {
+    const fetchUserDetails = async () => {
+      const token = localStorage.getItem('token');
+      try {
+        if (token) {
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:32]: @rahulg1254 remove this block, not required

File: src/pages/StudentBasicDetails.tsx (hunk 1 above)
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx:183]: resolved

File: src/pages/StudentBasicDetails.tsx [hunk 5]
```diff
@@ -114,41 +112,33 @@ const StudentDetails = () => {
                 justifyContent: 'center',
                 display: 'flex',
                 marginTop: 2
-              }}
-            >
+              }}>
               {renderStatsCard('Attendance', '78%')}
               {renderStatsCard('Classes Missed', '2')}
```
[the remark this answers is not in the document]
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx:117]: Added

File: src/pages/StudentBasicDetails.tsx (hunk 2 above)
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx:81]: Added

File: src/pages/StudentBasicDetails.tsx [hunk 6]
```diff
@@ -114,41 +112,33 @@ const StudentDetails = () => {
...
-        >
+          }}>
           <CardContent>
             <Typography
-              sx={{
-                color: theme.palette.warning['A200'],
-                fontSize: '16px',
-                fontWeight: 500
-              }}
+              sx={{ color: theme.palette.warning['A200'], fontSize: '16px', fontWeight: 500 }}
```
[the remark this answers is not in the document]
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx:131]: removed

**Pull request merged**